tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -2,12 +2,44 @@ package scala.scalanative
 package build
 
 import java.nio.file.{Files, Path, Paths}
+import scala.collection.mutable
 import scala.scalanative.util.Scope
-import scala.util.Try
 
 /** Utility methods for building code using Scala Native. */
 object Build {
 
+  // holds project basename and the related Config
+  private val cache = mutable.HashMap.empty[String, Config]
```

Comment:
To be honest I don't think caching Config itself is going to make any significant improvement. The only slow part of its creation is the `Discovery` for `NativeConfig`. There is no other IO or process invocation.  
The next possible IO operation is happening in the `Validator` by checking if each file on the classpath exists. In that case, it would be safer to ensure that all the files exist, and e.g. downloaded dependency was not removed between scalac compilation and the beginning of the linking. That would typically be equal to cost IO for `File.exists` invocation for each jar + a single `Files.isDirectory` compilation outputs of current project.

Reply:
Yes, I didn't remember how much time it took so I agree that it is not needed. I also wanted to see if the approach would be able to handle the scripted tests and such but clearly it doesn't. The last PR I was mixing too many things together. I think we could use the structure of the idea though to make the config naming better like is is now with the `buildImpl`.